### PR TITLE
Refactor Sass for better mobile support and maintainability

### PR DESCRIPTION
**Pull request by davidrleonard**
* Extracts the px-catalog app shell Sass from the giant predix-ui.scss file into its own, slimmed down file. This should help speed up time to interactive a bit because the app shell will only load/use the CSS it actually needs.
* Refactors the HTML structure and CSS structure for the app shell to make it more maintainable
* Fixes scrolling bugs for collapsed menu on mobile safari (#219)
* Fixes small flex iframe fit bugs (#82)
* Switches over to Google's app-drawer component for the side nav (woot)

@talimarcus please review
@mdwragg @randyaskin comments welcome

### CODE REVIEW

File: elements/px-catalog/px-catalog.html [hunk 1]
```diff
@@ -211,15 +201,36 @@
...
+         * Opens and closes the menu. The menu should remain open all the time
+         * on desktop-up viewports.
+         */
+        menuOpened: {
+          type: Boolean,
+          value: false
+        },
+
+        /**
+         * When the viewport width becomes larger than
```
Comment (talimarcus) [inline elements/px-catalog/px-catalog.html]: Larger than ... ?

File: elements/px-catalog/px-catalog.html [hunk 2]
```diff
@@ -211,15 +201,36 @@
...
+         * @type {Object}
+         */
+        isDesktopUp: {
+          type: Boolean,
+          observer: '_desktopBreakpointHit'
+        },
+
```
Comment (talimarcus) [inline elements/px-catalog/px-catalog.html:221]: add documentation here?

**Review by talimarcus — COMMENTED** [review]
Only comment on the Sass is that I know Kate has recommended using calculateRem(Xpx) rather than setting something to Xpx. Not sure if that is relevant here.

File: elements/px-catalog/px-catalog.html (hunk 1 above)
Comment (davidrleonard) [reply elements/px-catalog/px-catalog.html]: 👍  fixed.

File: elements/px-catalog/px-catalog.html (hunk 2 above)
Comment (davidrleonard) [reply elements/px-catalog/px-catalog.html:221]: 👍  fixed.

**Comment by davidrleonard** [comment]
@Menkhaus-ge this is ready if you give it the 👍

**Pull request merged**